osu.Game/Screens/Play/HUD/MultiplayerGameplayLeaderboard.cs

```diff
@@ -33,43 +48,68 @@ public MultiplayerGameplayLeaderboard(ScoreProcessor scoreProcessor, int[] userI
             this.scoreProcessor = scoreProcessor;
 
             // todo: this will likely be passed in as User instances.
-            this.userIds = userIds;
+            playingUsers = new BindableList<int>(userIds);
         }
 
-        [Resolved]
-        private SpectatorStreamingClient streamingClient { get; set; }
-
-        [Resolved]
-        private UserLookupCache userLookupCache { get; set; }
-
-        private Bindable<ScoringMode> scoringMode;
-
         [BackgroundDependencyLoader]
         private void load(OsuConfigManager config, IAPIProvider api)
         {
             streamingClient.OnNewFrames += handleIncomingFrames;
 
-            foreach (var user in userIds)
+            foreach (var userId in playingUsers)
             {
-                streamingClient.WatchUser(user);
+                streamingClient.WatchUser(userId);
 
                 // probably won't be required in the final implementation.
-                var resolvedUser = userLookupCache.GetUserAsync(user).Result;
+                var resolvedUser = userLookupCache.GetUserAsync(userId).Result;
 
                 var trackedUser = new TrackedUserData();
 
-                userScores[user] = trackedUser;
+                userScores[userId] = trackedUser;
                 var leaderboardScore = AddPlayer(resolvedUser, resolvedUser.Id == api.LocalUser.Value.Id);
 
                 ((IBindable<double>)leaderboardScore.Accuracy).BindTo(trackedUser.Accuracy);
                 ((IBindable<double>)leaderboardScore.TotalScore).BindTo(trackedUser.Score);
                 ((IBindable<int>)leaderboardScore.Combo).BindTo(trackedUser.CurrentCombo);
+                ((IBindable<bool>)leaderboardScore.HasQuit).BindTo(trackedUser.UserQuit);
             }
 
             scoringMode = config.GetBindable<ScoringMode>(OsuSetting.ScoreDisplayMode);
             scoringMode.BindValueChanged(updateAllScores, true);
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            // BindableList handles binding in a really bad way (Clear then AddRange) so we need to do this manually..
```

Comment:
side note - `BindableList.Clear()` should maybe be raising a `NotifyCollectionChangedAction.Reset` on clear, although that's debatable. sort of unwieldy indeed.